modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java

```diff
@@ -183,6 +190,45 @@ public void include(
 				unsupportedApplication = true;
 			}
 
+			Group group = themeDisplay.getScopeGroup();
+
+			boolean showContextChangePopover = false;
+
+			if (!CTCollectionThreadLocal.isProductionMode() && group.isSite()) {
+				HttpSession httpSession =
+					PortalSessionThreadLocal.getHttpSession();
+
+				long ctLastGroupId = GetterUtil.getLong(
+					httpSession.getAttribute("ctLastGroupId"));
+
+				if (ctLastGroupId == 0) {
+					ctLastGroupId = group.getGroupId();
+
+					httpSession.setAttribute("ctLastGroupId", ctLastGroupId);
+				}
+
+				if (ctLastGroupId != group.getGroupId()) {
+					httpSession.setAttribute("ctShowPopover", Boolean.TRUE);
+				}
+
+				if (GetterUtil.getBoolean(
+						httpSession.getAttribute("ctShowPopover"))) {
+
+					PanelCategoryHelper panelCategoryHelper =
+						new PanelCategoryHelper(_panelAppRegistry);
+
+					if (Validator.isNotNull(portletId) &&
+						panelCategoryHelper.containsPortlet(
+							portletId, PanelCategoryKeys.SITE_ADMINISTRATION) &&
+						!panelCategoryHelper.containsPortlet(
+							portletId,
+							PanelCategoryKeys.SITE_ADMINISTRATION_PUBLISHING)) {
+
+						showContextChangePopover = true;
+					}
+				}
+			}
+
```

Comment:
Move this to a private helper method to avoid cluttered code, so it's easier to read.